src/umpire/strategy/AllocationStrategy.hpp

```diff
@@ -74,15 +74,30 @@ class AllocationStrategy :
      *
      * \return Current total size of allocations.
      */
-    virtual long getCurrentSize() noexcept = 0;
+    virtual long getCurrentSize() const noexcept = 0;
 
     /*!
      * \brief Get the high watermark of the total allocated size.
      *
      * This is equivalent to the highest observed value of getCurrentSize.
      * \return High watermark allocation size.
      */
-    virtual long getHighWatermark() noexcept = 0;
+    virtual long getHighWatermark() const noexcept = 0;
+
+    /*!
+     * \brief Get the number of bytes that may be released back to resource
+     *
+     * A memory pool has a set of blocks that have no allocations
+     * against them.  If the size of the set is greater than one, then
+     * the pool will have a number of bytes that may be released back to
+     * the resource or coalesced into a larger block.
+     *
+     * Note: This method will always return 0 for strategies that are not
+     *       memory pools.
+     *
+     * \return The total number of bytes that are releaseable
+     */
+    virtual long getReleaseableSize() const noexcept;
```

Comment:
I thought this was only going to be added to the DynamicPool, since it's not applicable to multiple strategies?

Reply:
You're right.  Fixed.